app/controllers/search_controller.rb

```diff
@@ -184,19 +172,36 @@ def rate_limit_search
           "search-min",
           SiteSetting.rate_limit_search_user,
           1.minute,
+          error_code: 429,
         ).performed!
       else
         RateLimiter.new(
           nil,
-          "search-min-#{request.remote_ip}",
-          SiteSetting.rate_limit_search_anon_user,
+          "search-min-#{request.remote_ip}-per-sec",
+          SiteSetting.rate_limit_search_anon_user_per_second,
+          1.second,
+          error_code: 429,
+        ).performed!
+        RateLimiter.new(
+          nil,
+          "search-min-#{request.remote_ip}-per-min",
+          SiteSetting.rate_limit_search_anon_user_per_minute,
           1.minute,
+          error_code: 429,
```

Comment:
do we need to specify error code given: https://github.com/discourse/discourse/blob/df9b0255d8dc14a6de6d5baea87a333f3cbed7ae/app/controllers/application_controller.rb#L191-L191 ?